rustls/src/quic.rs

```diff
@@ -54,23 +92,18 @@ pub trait QuicExt {
     /// Emit unencrypted TLS handshake data.
     ///
     /// When this returns `Some(_)`, the new keys must be used for future handshake data.
```

Comment:
I have trouble understanding how `KeyMaterial` is to be used and I think it would be good to clarify how to use each variant. I think `KeyMaterial`'s and/or this function's documentation needs to be updated/expanded to explain how to use them, especially so people maintaining Rustls can understand it better.

Reply:
So we have multiple things we want to return. When the handshake has started, we return handshake keys. These include header and packet keys, and we don't need to do key updates for these keys. When traffic can start, we get a new set of header and packet keys. Then, we also need to get a set of `Secrets` so we can derive future key updates when either the peer or us finds it necessary. Key updates only update the packet keys, not the header keys, so instead of just returning the `Secrets` from which the first set of 1-RTT keys can be derived, we return that full set (including the header protection keys), and yield the `Secrets` to be used for the first key update.

Reply:
I added a longer comment for the type now named `KeyChange`.